File: api/rating-service/src/main/java/com/carrental/ratingservice/model/dto/CommentWithRateAddDTO.java
package com.carrental.ratingservice.model.dto;

import lombok.Data;

import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

@Data
public class CommentWithRateAddDTO {

    @NotNull
    private String content;

    @NotNull
    private Long vehicleId;

    @NotNull
    private Long userId;

    @NotNull
    @Min(value = 1)
    @Max(value = 5)
    private int rate;
}

File: frontend-app/main-page/src/model/CommentWithRateAddDTO.ts
export class CommentWithRateAddDTO {
    rate: number;
    content: string;
    vehicleId: string;
    userId: string;

    constructor(rate: number, content: string, vehicleId: string, userId: string) {
        this.rate = rate;
        this.content = content;
        this.vehicleId = vehicleId;
        this.userId = userId;
    }
}
